osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/BatchInserter.cs

```diff
@@ -0,0 +1,428 @@
+                        {
+                            Console.WriteLine($"Legacy score {highScore.score_id} has invalid date ({highScore.date}), fixing.");
+                            highScore.date = DateTimeOffset.UnixEpoch;
+                        }
+
+                        SoloScoreLegacyIDMap? existingMapping = existingIds.FirstOrDefault(e => e.old_score_id == highScore.score_id);
+
+                        if ((existingMapping != null && skipExisting) || (existingMapping == null && skipNew))
+                        {
+                            Interlocked.Increment(ref TotalSkipCount);
+                            continue;
+                        }
+
+                        ScoreInfo referenceScore = await createReferenceScore(ruleset, highScore, db, transaction);
+                        string serialisedScore = JsonConvert.SerializeObject(new SoloScoreInfo
+                        {
+                            // id will be written below in the UPDATE call.
+                            UserID = highScore.user_id,
+                            BeatmapID = highScore.beatmap_id,
+                            RulesetID = ruleset.RulesetInfo.OnlineID,
+                            Passed = true,
+                            TotalScore = (int)referenceScore.TotalScore,
+                            Accuracy = referenceScore.Accuracy,
+                            MaxCombo = highScore.maxcombo,
+                            Rank = Enum.TryParse(highScore.rank, out ScoreRank parsed) ? parsed : ScoreRank.D,
+                            Mods = referenceScore.Mods.Select(m => new APIMod(m)).ToArray(),
+                            Statistics = referenceScore.Statistics,
+                            MaximumStatistics = referenceScore.MaximumStatistics,
+                            EndedAt = highScore.date,
+                            LegacyTotalScore = highScore.score,
+                            LegacyScoreId = highScore.score_id
+                        }, new JsonSerializerSettings
+                        {
+                            DefaultValueHandling = DefaultValueHandling.Ignore
+                        });
+
+                        if (existingMapping != null)
+                        {
+                            if (highScore.is_deletion)
+                            {
+                                deleteCommand.Transaction = transaction;
+
+                                deleteOldId.Value = existingMapping.old_score_id;
+                                deleteNewId.Value = existingMapping.score_id;
+
+                                if (!deleteCommand.IsPrepared)
+                                    await deleteCommand.PrepareAsync();
+
+                                await runCommand(deleteCommand);
+                                await enqueueForFurtherProcessing(existingMapping.score_id, db, transaction, true);
+
+                                // TODO: delete
```

Comment:
What does this todo mean?

Reply:
implement deletion, but i guess i did elsewhere